spec/inputs/couchdb_changes_spec.rb

```diff
@@ -1 +1,507 @@
+    index = "couchdb_test"
+
+    before do
+      buildup
+      deletedoc # from CouchDB
+    end
+    
+    ftw = FTW::Agent.new
+
+    config <<-CONFIG
+    input {
+      couchdb_changes {
+        db => "db"
+        host => "127.0.0.1"
+        timeout => 2000
+        always_reconnect => false
+        sequence_path => "#{sequence}"
+        type => "couchdb"
+      }
+    }
+    output {
+      elasticsearch {
+        action => "%{[@metadata][action]}"
+        document_id => "%{[@metadata][_id]}"
+        host => "127.0.0.1"
+        index => "#{index}"
+        protocol => "http"
+      }
+    }
+    CONFIG
+    
+    agent do
+      # Verify the count (should now be 9)
+      ftw.post!("http://127.0.0.1:9200/#{index}/_refresh")
+      data = ""
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_count?q=*")
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      count = result["count"]
+      insist { count } == 9
+      # Get the docs and do a couple more spot checks
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_search?q=*&size=10")
+      data = ""
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      found = false
+      result["hits"]["hits"].each do |doc|
+        if doc["_id"] == "9"
+          found = true
+        end
+      end
+      insist { found } == false
+    end
+    
+    after do
+      teardown
+    end
+
+  end
+
+  describe "Test authenticated connectivity", :elasticsearch => true, :couchdb => true do
+    include Helpers
+    user = "logstash"
+    pass = "logstash"
+    sequence = "/tmp/.couchdb_seq"
+    index = "couchdb_test"
+
+    before do
+      buildup
+      createuser
+    end
+
+    ftw = FTW::Agent.new
+
+    config <<-CONFIG
+    input {
+      couchdb_changes {
+        db => "db"
+        host => "127.0.0.1"
+        timeout => 2000
+        always_reconnect => false
+        sequence_path => "#{sequence}"
+        type => "couchdb"
+        username => "#{user}"
+        password => "#{pass}"
+      }
+    }
+    output {
+      elasticsearch {
+        action => "%{[@metadata][action]}"
+        document_id => "%{[@metadata][_id]}"
+        host => "127.0.0.1"
+        index => "#{index}"
+        protocol => "http"
+      }
+    }
+    CONFIG
+      
+    agent do
+      # Verify the count
+      ftw.post!("http://127.0.0.1:9200/#{index}/_refresh")
+      data = ""
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_count?q=*")
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      count = result["count"]
+      insist { count } == 10
+      # Get the docs and do a couple spot checks
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_search?q=*&size=10")
+      data = ""
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      result["hits"]["hits"].each do |doc|
```

Comment:
This misses the case a doc with _id == 3 is not in results.
You can write this as:

```ruby
doc3 = result["hits"]["hits"].find { |doc| doc["_id"] == 3 }

# Make sure it's found
reject { doc3 }.nil?
# verify the 'name' field
insist { doc3["_source"]["name"] } == "Captain America"
```